java/arcs/core/data/Schema.kt

```diff
@@ -15,14 +15,27 @@ import arcs.core.crdt.CrdtEntity
 import arcs.core.crdt.VersionMap
 import arcs.core.type.Type
 
+/** Returns true if the RawEntity data matches the refinement predicate */
+typealias Refinement = (data: RawEntity) -> Boolean
+
+/** Returns true if the RawEntity data matches the query predicate (given a query argument)*/
+typealias Query = (data: RawEntity, query_arg: Any) -> Boolean
+
 data class Schema(
     val names: List<SchemaName>,
     val fields: SchemaFields,
     /**
      * The hash code for the schema (note that this is not the same as this object's [hashCode]
      * method.
      */
-    val hash: String
+    val hash: String,
+    val refinement: Refinement = { _: RawEntity -> true },
+    val query: Query = {
+        _: RawEntity, _: Any ->
+            throw IllegalArgumentException(
```

Comment:
Maybe use `NotImplementedException`... or better yet: have `val query` be of type. `Query?`, and use `requireNotNull` when you call it.